src/components/ArticleCreationModal.jsx

```diff
@@ -0,0 +1,98 @@
+import axios from 'axios';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { Button, Modal, Form, Segment } from 'semantic-ui-react';
+
+const ArticleCreationModal = () => {
+  const [open, setOpen] = useState(false);
+  const [category, setCategory] = useState();
+  const dispatch = useDispatch();
+
+  const categories = [
+    { key: 'FE', text: 'Flat Earth', value: 'flatEarth' },
+    { key: 'UFO', text: 'Aliens', value: 'aliens' },
+  ];
+
+  const createArticle = async (event) => {
+    event.preventDefault();
+    try {
+      let response = await axios.post('/articles', {
+        params: {
+          title: event.target.title.value,
+          category: category,
+          teaser: event.target.teaser.value,
+          body: event.target.body.value,
+        },
+      });
+      dispatch({ type: 'SUCCESS_MESSAGE', payload: response.data.message });
+    } catch (error) {
+      if (error.response.status === 500) {
+        dispatch({
+          type: 'ERROR_MESSAGE',
+          payload: 'Something went wrong on our server, try again later',
+        });
+      } else {
+        dispatch({ type: 'ERROR_MESSAGE', payload: error.message });
+      }
+    }
+    setOpen(false);
+  };
```

Comment:
I would keep the remote CRUD actions out of the components. I would suggest creating a service module that deals with that.

Reply:
CRUD action now moved to modules folder